### DISCUSSION TITLE
Installing optional packages

### DISCUSSION THREAD

**Discussion by BriacD** [netbox-community/netbox#14379] [Help Wanted!] [answered]
I'm attempting to set up a git repo as a remote data source. I noticed in the documentation that the required packaged for this, dulwich, is no longer included in 3.6>

I'm having trouble understanding how to install and set up references to the package in question.

Per the documentation: _The boto3 and dulwich packages are no longer installed automatically. If needed for S3/git remote data backend support, add them to local_requirements.txt to ensure their installation._

I've installed the packed via this documentation:  https://www.dulwich.io/docs/
and updated my local_requirements via this command: sudo sh -c "echo 'dulwich' >> /opt/netbox/local_requirements.txt"

Yet my data source is still failing to sync. I do not know if there is a way to find some logs on the failure but obviously I'm doing something wrong.

Any help would be appreciated.

**Comment by abhi1693** [comment]
Did you restart netbox services after the install?

**Reply by BriacD** [reply]
Sorry. I did. Forgot to mention that.

**Comment by jeremystretch** [comment, marked as answer]
You need to run the upgrade script again after adding the package to ~`requirements.txt`~ `local_requirements.txt`. If it's still not working, verify that `dulwich` is actually getting installed in the NetBox virtual environment.

**Reply by BriacD** [reply]
Gotcha. 
Just so I understand, I also need to include the reference within _/opt/netbox/requirements.txt_ as well as _/opt/netbox/local_requirements.txt_ ?

**Reply by BriacD** [reply]
Scratch that. I do not it would seem. It synced!

Thanks for the quick and helpful response.

**Reply by jeremystretch** [reply]
Sorry, I meant `local_requirements.txt`, not `requirements.txt`! You should never need to modify `requirements.txt` (although `pip` will install packages listed in either file).

**Reply by BriacD** [reply]
Understood. Thanks again.
